### PR TITLE
Added custom hooks for withdraw liquidity, repay loan and open loan

### PR DESCRIPTION
**Pull request by UsamaSaleem14**

### CODE REVIEW

**Review by TBomberman — COMMENTED** [review]
ok with me @Pochetes

File: hooks/useWithdrawLiquidityHandler.tsx [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
+import IERC20Metadata from '../abis/IERC20Metadata.json'
+import {sqrt} from '../utils/mathFunctions'
+import Tokens, {Token} from '../components/Tokens'
 
 const ZEROMIN = 0
 
-const WithdrawLiquidity = () => {
+export const useWithdrawLiquidityHandler = () => {
   const [depPool, setdepPool] = useState<Contract | null>(null)
   const [sliderPercentage, setsliderPercentage] = useState(0)
```
Comment (Pochetes) [inline hooks/useWithdrawLiquidityHandler.tsx:16]: this should be typed  of type `number`

File: hooks/useWithdrawLiquidityHandler.tsx [hunk 2]
```diff
@@ -59,6 +64,7 @@ const WithdrawLiquidity = () => {
         setenableRemove(true)
       })
       .catch((err: any) => {
+        setenableRemove(false)
```
Comment (Pochetes) [inline hooks/useWithdrawLiquidityHandler.tsx:67]: nit: camelcase this function

File: pages/positions/openloan.tsx [hunk 3]
```diff
@@ -1,10 +1,295 @@
...
+import PairsSelector from '../../components/PairsSelector'
+import {useOpenLoanHandler} from '../../hooks/useOpenLoanHandler'
+
+const style = {
+  wrapper: 'w-screen flex justify-center items-center',
+  formContent: 'bg-gray-900 w-[30rem] rounded-2xl p-4 max-w-lg',
+  vStack: 'items-center flex-col',
+  vStackItem: 'mt-3',
+  formHeader: 'justify-between items-center font-semibold text-xl text-gray-200 text-center',
+  numberInputContainer: 'bg-gray-800 rounded-2xl p-4 border-2 border-gray-800 hover:border-gray-600 flex justify-between w-full',
```
Comment (Pochetes) [inline pages/positions/openloan.tsx:19]: if the styling starts getting this big, it's best to separate it to its own file

File: pages/positions/openloan.tsx [hunk 4]
```diff
@@ -1,10 +1,295 @@
...
+  const [collateral1Class, setCollateral1Class] = useState<string>('')
+
+  useEffect(() => {
+    resetCollateralType()
+    validate()
+  }, [token0, token1])
+
+  useEffect(() => {
+    setIsOpen(false)
+    // console.log("selected collateral type", CollateralType[collateralType])
```
Comment (Pochetes) [inline pages/positions/openloan.tsx:73]: nit: remove `console.log`

File: pages/positions/openloan.tsx [hunk 5]
```diff
@@ -1,10 +1,295 @@
...
+    setCollateralAmt1Str('')
+    setCollateral1Class(collateralType == CollateralType.Both ? style.numberInputContainer : style.numberInputHidden)
+    validate()
+  }, [collateralType])
+
+  function getCollateralTypeButtonText(collateralType: CollateralType) {
+    switch (collateralType) {
+      case CollateralType.None:
+        return 'Select collateral type'
+      case CollateralType.LPToken:
```
Comment (Pochetes) [inline pages/positions/openloan.tsx:85]: there are some functions that need to go to the custom hook, that's what it's used for. If it gets too big, you can also separate them into smaller, more atomic custom hooks.

File: pages/positions/openloan.tsx [hunk 6]
```diff
@@ -1,10 +1,295 @@
...
+      if (error instanceof Error) message = error.message
+      else message = String(error)
+
+      notifyError(message)
+    }
+  }
+
+  setValue('collateralType', collateralType)
+  return (
+    <>
```
Comment (Pochetes) [inline pages/positions/openloan.tsx:198]: is this needed? I feel like you can remove it and it'll still be fine.

**Review by Pochetes — APPROVED** [review]
LGTM, make sure to review comments

File: pages/positions/openloan.tsx (hunk 6 above)
Comment (UsamaSaleem14) [reply pages/positions/openloan.tsx:198]: In react a component must return only single element. By using the empty tag syntax one doesn’t have to either import React or write the long keyword React.Fragment every time. So it’s basically a Syntactic sugar which makes our lives easier.

Comment (UsamaSaleem14) [reply pages/positions/openloan.tsx]: Done

File: pages/positions/openloan.tsx (hunk 5 above)
Comment (UsamaSaleem14) [reply pages/positions/openloan.tsx]: Done

File: pages/positions/openloan.tsx (hunk 4 above)
Comment (UsamaSaleem14) [reply pages/positions/openloan.tsx]: Done

File: pages/positions/openloan.tsx (hunk 3 above)
Comment (UsamaSaleem14) [reply pages/positions/openloan.tsx]: Done

File: hooks/useWithdrawLiquidityHandler.tsx (hunk 2 above)
Comment (UsamaSaleem14) [reply hooks/useWithdrawLiquidityHandler.tsx]: Done

File: hooks/useWithdrawLiquidityHandler.tsx (hunk 1 above)
Comment (UsamaSaleem14) [reply hooks/useWithdrawLiquidityHandler.tsx]: Done

File: hooks/useOpenLoanHandler.tsx [hunk 7]
```diff
@@ -34,32 +33,28 @@ const OpenLoanController = () => {
...
-      BigNumber.from(0)
-    var amt1BN = data.collateralAmt1 ?
-      ethers.utils.parseUnits(data.collateralAmt1, token1.decimals) :
-      BigNumber.from(0)
+    var amt0BN = data.collateralAmt0 ? ethers.utils.parseUnits(data.collateralAmt0, token0.decimals) : BigNumber.from(0)
+    var amt1BN = data.collateralAmt1 ? ethers.utils.parseUnits(data.collateralAmt1, token1.decimals) : BigNumber.from(0)
     var loanAmtBN = ethers.utils.parseUnits(data.loanAmt, 18) // 18 is from DepositPool.decimals
     var collateralType = data.collateralType
-    var accountAddress = accountInfo.address ? accountInfo.address : ""
+    var accountAddress = accountInfo.address ? accountInfo.address : ''
```
[the remark this answers is not in the document]
Comment (UsamaSaleem14) [reply hooks/useOpenLoanHandler.tsx]: Done

**Pull request merged**